steampipecloud/table_steampipecloud_workspace_aggregator.go

```diff
@@ -0,0 +1,495 @@
+	if err != nil {
+		plugin.Logger(ctx).Error("getOrgWorkspaceAggregator", "get", err)
+		return nil, err
+	}
+
+	aggregator = response.(openapi.WorkspaceAggregator)
+
+	return aggregator, nil
+}
+
+func getIdentityWorkspaceDetailsForAggregator(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	// Create Session
+	svc, err := connect(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("getIdentityWorkspaceDetailsForAggregator", "connection_error", err)
+		return nil, err
+	}
+
+	var identityWorkspaceDetails IdentityWorkspaceDetailsForAggregator
+	// get workspace details from hydrate data
+	// workspace details reside in the parent item in this case
+	switch w := h.ParentItem.(type) {
+	case openapi.Workspace:
+		plugin.Logger(ctx).Debug("getIdentityWorkspaceDetailsForAggregator", "openapi.Workspace")
+		identityId := h.ParentItem.(openapi.Workspace).IdentityId
+		identityWorkspaceDetails.WorkspaceHandle = h.ParentItem.(openapi.Workspace).Handle
+		getDetails := func(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+			if strings.HasPrefix(identityId, "u_") {
+				resp, _, err := svc.Users.Get(ctx, identityId).Execute()
+				identityWorkspaceDetails.IdentityType = "user"
+				identityWorkspaceDetails.IdentityHandle = resp.Handle
+				return nil, err
+			} else {
+				resp, _, err := svc.Orgs.Get(ctx, identityId).Execute()
+				identityWorkspaceDetails.IdentityType = "org"
+				identityWorkspaceDetails.IdentityHandle = resp.Handle
+				return nil, err
+			}
+		}
+		_, _ = plugin.RetryHydrate(ctx, d, h, getDetails, &plugin.RetryConfig{ShouldRetryError: shouldRetryError})
+		return identityWorkspaceDetails, nil
+	case *openapi.Workspace:
+		plugin.Logger(ctx).Debug("getIdentityWorkspaceDetailsForAggregator", "*openapi.Workspace")
+		identityId := h.ParentItem.(*openapi.Workspace).IdentityId
+		identityWorkspaceDetails.WorkspaceHandle = h.ParentItem.(*openapi.Workspace).Handle
+		getDetails := func(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+			if strings.HasPrefix(identityId, "u_") {
+				resp, _, err := svc.Users.Get(ctx, identityId).Execute()
+				identityWorkspaceDetails.IdentityType = "user"
+				identityWorkspaceDetails.IdentityHandle = resp.Handle
+				return nil, err
+			} else {
+				resp, _, err := svc.Orgs.Get(ctx, identityId).Execute()
+				identityWorkspaceDetails.IdentityType = "org"
+				identityWorkspaceDetails.IdentityHandle = resp.Handle
+				return nil, err
+			}
+		}
+		_, _ = plugin.RetryHydrate(ctx, d, h, getDetails, &plugin.RetryConfig{ShouldRetryError: shouldRetryError})
+		plugin.Logger(ctx).Debug("getIdentityWorkspaceDetailsForAggregator", "identityWorkspaceDetails", identityWorkspaceDetails)
+		return &identityWorkspaceDetails, nil
+	default:
+		plugin.Logger(ctx).Debug("getIdentityWorkspaceDetailsForAggregator", "Unknown Type", w)
```

Comment:
@siddharthaturbot could you please add some details/comments around the `Unknown Type`?

Reply:
Yes @siddharthaturbot, I think additional comments would be helpful. Thanks for the above explanation.